### PR TITLE
Have codegen log status diffs at debug level

### PR DESCRIPTION
**Pull request by mattmoor**
Fixes: https://github.com/knative/pkg/issues/1597

/hold

For some eyeballs

### CODE REVIEW

File: client/injection/apiextensions/reconciler/apiextensions/v1beta1/customresourcedefinition/reconciler.go [hunk 1]
```diff
@@ -317,6 +318,10 @@ func (r *reconcilerImpl) updateStatus(existing *v1beta1.CustomResourceDefinition
 			return nil
 		}
 
+		if diff, err := kmp.SafeDiff(existing.Status, desired.Status); err == nil && diff != "" {
+			logging.FromContext(ctx).Debugf("Updating status with: %s", diff)
```
Comment (whaught) [inline client/injection/apiextensions/reconciler/apiextensions/v1beta1/customresourcedefinition/reconciler.go:322]: We may want to include the GVK or some identifier for the resource in the message?

Comment (mattmoor) [reply client/injection/apiextensions/reconciler/apiextensions/v1beta1/customresourcedefinition/reconciler.go:322]: The logger is already decorated with the key, and this should print sufficient enclosing context about the Go type to determine the type information.  Maybe we should decorate the logger with `gvk: <foo>` independent of this if that's useful...?

Comment (mattmoor) [reply client/injection/apiextensions/reconciler/apiextensions/v1beta1/customresourcedefinition/reconciler.go:322]: FYI the key decoration is generally how logstream works, which is a big reason I want this :)

Comment (whaught) [reply client/injection/apiextensions/reconciler/apiextensions/v1beta1/customresourcedefinition/reconciler.go:322]: /lgtm

**Comment by mattmoor** [comment]
/unhold

I'm gonna try to get this into serving to see how it looks in the logstream 😇

**Pull request merged**